# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/TextDirectiveUtil.cpp
# Commit: 851ad0761b61
# Full Hash: 851ad0761b619ebf4b4982a18ab0bb627d7dd240
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-19 09:41:31
# Regressor Bug: 1976473
# File Overlap Count: 2
# Description:
#   Bug 1976473 - Text Fragments: Treat punctuation as part of a word when computing word boundary distances. r=dom-core,avandolder,mccr8
#   
#   Finding word boundaries uses `intl::WordBreaker::FindWord()`, which stops at punctuation.
#   As a result, finding word boundary distances for the string "foo bar. baz" would have had found
#   the distances `[3,7,8,12]` when going LTR and `[3,5,8,12]` when going RTL.
# ==============================================================================

diff -r 6a6745867aa0 -r 851ad0761b61 dom/base/TextDirectiveUtil.cpp
--- a/dom/base/TextDirectiveUtil.cpp	Fri Jul 18 18:12:34 2025 +0000
+++ b/dom/base/TextDirectiveUtil.cpp	Fri Jul 18 18:16:32 2025 +0000
@@ -254,4 +254,16 @@
   return {node, pos};
 }
 
+/* static */ bool TextDirectiveUtil::WordIsJustWhitespaceOrPunctuation(
+    const nsAString& aString, uint32_t aWordBegin, uint32_t aWordEnd) {
+  MOZ_ASSERT(aWordEnd <= aString.Length());
+  MOZ_ASSERT(aWordBegin < aWordEnd);
+
+  auto word = aString.View().substr(aWordBegin, aWordEnd - aWordBegin);
+  return std::all_of(word.cbegin(), word.cend(), [](const char16_t ch) {
+    return nsContentUtils::IsHTMLWhitespaceOrNBSP(ch) ||
+           mozilla::IsPunctuationForWordSelect(ch);
+  });
+}
+
 }  // namespace mozilla::dom